### ISSUE TITLE
Note refactoring support

### ISSUE THREAD

**Issue by kevinslin** [dendronhq/dendron#39]
## Context

Support automatically renaming files and links.

## Proposal

- create `> Dendron: Refactor Hierarchy Command`
- algo
  - take regex as an input that includes a `match` pattern and `replace` pattern
  - gather all files that match regex and rename `match` to `replace`
  - gather all files with backlinks to renamed files and update links 

### Example

- original file structure
```
.
├── pro.foo.md
├── pro.foo.child.md
├── pro.foo.child.grandchild.mmd
└── bar.foo.md  # has link to [[pro.foo]]
```

- after running `> Dendron: Refactor Hierarchy` with `s/^pro//`
```
.
├── foo.md
├── foo.child.md
├── foo.child.grandchild.mmd
└── bar.foo.md  # has link to [[foo]]
```

**Comment by kpathakota** [comment]
Just adding more context for the way I'd use it in case it would help - I have a projects path and an archive path and when I'm done with an active project, I'd like a way to move all of them into the archive path. 

Link renaming is a close second.

**Comment by kevinslin** [comment]
shipped in 4fcaf40
